File: lib/core/src/main/java/io/loglens/core/Level.java
package io.loglens.core;

/** 로그 레벨. 문자 하나가 곧 레코드에 찍히는 값이다. */
public enum Level {
    V('V'), D('D'), I('I'), W('W'), E('E');

    public final char c;

    Level(char c) {
        this.c = c;
    }

    /**
     * 릴리스 게이팅: V/D 는 디버그 빌드에서만, I/W/E 는 항상.
     * 기획서 §5 — 라이브러리는 앱의 빌드타입을 모르므로 {@code isDebug} 를 주입받는다.
     */
    public boolean enabled(boolean isDebug) {
        return isDebug || ordinal() >= I.ordinal();
    }
}

File: lib/core/bin/main/io/loglens/core/Level.kt
package io.loglens.core

/** 로그 레벨. 문자 하나가 그대로 로그에 찍힙니다. */
enum class Level(@JvmField val c: Char) {
    V('V'), D('D'), I('I'), W('W'), E('E');

    /**
     * 릴리스에서 V/D 를 막습니다. I/W/E 는 항상 통과합니다.
     *
     * 라이브러리는 앱의 빌드 타입을 알 수 없으므로 [isDebug] 를 주입받습니다.
     */
    fun enabled(isDebug: Boolean): Boolean = isDebug || ordinal >= I.ordinal
}
